applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -80,11 +80,22 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateRightH
 {
     const TransonicPerturbationPotentialFlowElement& r_this = *this;
     const int wake = r_this.GetValue(WAKE);
+    const int kutta = r_this.GetValue(KUTTA);
 
     if (wake == 0) // Normal element (non-wake) - eventually an embedded
     {
-        CalculateRightHandSideNormalElement(rRightHandSideVector, rCurrentProcessInfo);
-
+        if (this->Is(INLET))
+        {
+            CalculateRightHandSideNormalElement(rRightHandSideVector, rCurrentProcessInfo);
+        }
+        else if(kutta != 0)
+        {
+            CalculateRightHandSideNormalElement(rRightHandSideVector, rCurrentProcessInfo);
+        }
+        else
+        {
+            CalculateRightHandSideExtendedElement(rRightHandSideVector, rCurrentProcessInfo);
+        }
```

Comment:
I like the idea of making a new function for the RHS. However, I see that, for now, inside CalculateRightHandSideExtendedElement most of the lines are the same as in CalculateRightHandSideNormalElement. For that reason I would keep just one function to avoid duplicated code and add it later if we need it. (I think that for the RHS it might be possible to use the same implementation for inlet and for the extended elements). See my suggestion inside CalculateRightHandSideNormalElement to make this work for all cases without duplicating code.
```suggestion
            if (rRightHandSideVector.size() != TNumNodes + 1) {
                rRightHandSideVector.resize(TNumNodes + 1, false);
            }
        } else {
            if (rRightHandSideVector.size() != TNumNodes) {
                rRightHandSideVector.resize(TNumNodes, false);
            }
        }
        rRightHandSideVector.clear();
        CalculateRightHandSideNormalElement(rRightHandSideVector, rCurrentProcessInfo);
```